v0/destinations/klaviyo/transform.js

```diff
@@ -100,58 +47,74 @@ const addUserToList = async (message, traitsInfo, conf, destination) => {
 // ---------------------
 const identifyRequestHandler = async (message, category, destination) => {
   // If listId property is present try to subscribe/member user in list
+  if (!destination.Config.privateApiKey) {
+    throw new CustomError(
+      "Private API Key is a required field for identify events",
+      400
+    );
+  }
   const traitsInfo = getFieldValueFromMessage(message, "traits");
-  if (
-    (!!destination.Config.listId || !!get(traitsInfo.properties, "listId")) &&
-    destination.Config.privateApiKey
-  ) {
-    await addUserToList(message, traitsInfo, LIST_CONF.MEMBERSHIP, destination);
-    if (get(traitsInfo.properties, "subscribe") === true) {
-      await addUserToList(
-        message,
-        traitsInfo,
-        LIST_CONF.SUBSCRIBE,
-        destination
+  const response = defaultRequestConfig();
+  const personId = await isProfileExist(message, destination);
+  if (!personId) {
+    if (!destination.Config.publicApiKey) {
+      throw new CustomError(
+        "Public API Key is a required field for identify events",
+        400
       );
     }
-  } else {
-    logger.info(
-      `Cannot process list operation as listId is not available, either in message or config, or private key not present`
+    const mappedToDestination = get(message, MappedToDestinationKey);
+    if (mappedToDestination) {
+      addExternalIdToTraits(message);
+      adduserIdFromExternalId(message);
+    }
```

Comment:
This should be done before we are doing look-up at line 58. For RETL source we are mapping `externalId` to `userId` which is important during look-up and subsequent flow.

Reply:
done